doc/articles/contributing/build-troubleshooting.md

```diff
@@ -0,0 +1,50 @@
+# build troubleshooting
+
+Head to Tools -> Options -> MS Build Verbosity
+
+- Change to detailed (window/console logger)
+- Change to detailed (log)
+
+Quit visual studio and ensure that all processes have been terminated.
+
+```bash
+taskkill /fi "imagename eq devenv.exe" /f /t
+taskkill /fi "imagename eq msbuild.exe" /f /t
+taskkill /fi "imagename eq Uno.SourceGeneration.Host.exe" /f /t
+```
+
+Reset the source generation cache, your nuget cache and all local changes to your git repository. Be careful, as this will wipe out code you haven't yet commited.
+
+```bash
+$ git reset --hard
+$ git clean -fdx
+$ git checkout master
+$ git pull origin master
+$ nuget locals all -clear
```

Comment:
`nuget locals all -clear` should be a last resort. Latest VS and nuget builds don't mess the cache anymore. Rebuilding it consumes a lot of time.

Reply:
Moved it down to last resort section.